### PR TITLE
New methods

### PR DESCRIPTION
**Pull request by ma-zal**
I implemented new methods:
- getFlowsLibrary
- filterRunnables
- getFlowDetails
- getFlowInputs
- getFlowOutputs
- getFlowSchedules
- getExecutions

that extends the current HP-OO API coverage.

### CODE REVIEW

File: lib/Executions.js [hunk 1]
```diff
@@ -22,6 +21,41 @@ class Executions {
...
+
+  /**
+   * Returns a list of executions summary, with filtering.
+   *
+   * @returns {Promise.<Execution[]>} The returned objects include the execution summary,
+   *                                  detailed objects with data about the execution.
+   */
+  async getExecutions () {
+    return this.requestManager.request({
+      actionUrl: this.executionsActionUrl + '?pageSize=1000&pageNum=1',
```
Comment (lirantal) [inline lib/Executions.js:33]: maybe this function can support parameters that will control the `pageSize`, `pageNum` ?
we can leave that for a follow-up PR though, it's not a must for this.

File: lib/Executions.js [hunk 2]
```diff
@@ -22,6 +21,41 @@ class Executions {
...
+      method: 'GET'
+    })
+  }
 }
 
 module.exports = Executions
+
+/**
+ * @typedef {Object} Execution
```
Comment (lirantal) [inline lib/Executions.js:42]: this for typescript or something?

File: lib/Schedules.js [hunk 3]
```diff
@@ -26,6 +26,14 @@ class Schedules {
 
     return result
   };
+
+  async getFlowSchedules () {
+    console.log('Loading shcedules')
```
Comment (lirantal) [inline lib/Schedules.js]: might want to remove this

**Review by lirantal — CHANGES_REQUESTED** [review]

**Comment by lirantal** [comment]
@ma-zal this is amazing! great work and thanks for contributing this.
really small change to remove the console.log out of the code-base and I'll merge.

200w_d 2


I also officially added you as a contributor to the project so you can review/merge PRs yourself as well! 💪

File: lib/Schedules.js (hunk 3 above)
Comment (ma-zal) [reply lib/Schedules.js]: Removed. Thanks.

File: lib/Executions.js (hunk 2 above)
Comment (ma-zal) [reply lib/Executions.js:42]: Yes, it is a JSdoc type definition, something like an interface in Typescript (if we have no Typescript) :)

**Comment by ma-zal** [comment]
Sorry @lirantal for forgetting this PR (for 6 years, I know). This repo looks like not maintained anymore. But as "nice to have" I am finishing this change. :-)

**Comment by ma-zal** [comment]
Note: I have no write permission to merge the change, so please merge or close this PR unmerged. Thanks, @lirantal.

**Review by lirantal — APPROVED** [review]

**Comment by lirantal** [comment]
Thanks!

**Comment by ma-zal** [comment]
Please merge the PR, @lirantal.
I have no writing permission for merging.

**Comment by lirantal** [comment]
Merged!

**Pull request merged**